### DISCUSSION TITLE
Would you mind introducing Headers.get_raw and Headers.raw_items methods?

### DISCUSSION THREAD

**Discussion by Colin-b** [encode/httpx#2841] [General]
Right now, the Headers class exposes (amongst others) the following methods:

# Headers.get
```python
def get(discarded_case_name: str) -> str:
  ...
```

If I want to look within the headers for a header, without matching case, I can use Headers.get and it will also handle multi values headers by joining the values with ", ".
However, if I want to do the same while matching casing, I have to iterate myself on Headers.raw property and also handle the multi values.
Would it be ok to introduce get_raw(case_respected_name: bytes) -> bytes ?

# Headers.items
```python
def items() -> Mapping[str, str]:
  ...
```

Following the same logic, I would like to browse through all the items, but with the actual name and multi values handling.
Would it be ok to introduce raw_items() -> Mapping[bytes, bytes] ?

**Comment by commonism** [comment]
Hi,

using "," does not work for cookies, as multiple cookies set via header are joined by ",", while multiple cookies set via cookies= are joined using ";".
